docs/source/blogs/tech_blog/blog12_Inference_Time_Compute_Implementation_in_TensorRT-LLM.md

```diff
@@ -0,0 +1,425 @@
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            self.generation_dir,
+            legacy=False,
+            padding_side='left',
+            truncation_side='left',
+            trust_remote_code=False,
+            use_fast=True,
+        )
+        self.streaming = streaming
+```
+
+The `process()` function, as mentioned before, is the core method within the `Controller` class. Here, we can customize our data flow by specifying the logic for yielding tasks. For Dynasor-CoT, we have two different kinds of tasks:
+
+1. `proposer_task`: Handles the main content generation, producing self.chunk_size tokens based on the previous content.
+
+2. `probe_task`: Elicits an early-stage answer by generating 20 tokens from the same content.
+
+The code below creates these two types of tasks.
+
+```python
+    def process(self, tasks: List[GenerationTask], **kwargs):
+        # Start with the initial prompt provided by the first task.
+        initial_prompt = tasks[0].input_str
+
+        proposer_task = GenerationTask()
+        proposer_task.max_tokens = self.chunk_size
+        proposer_task.temperature = 0.6
+        proposer_task.top_p = 0.95
+        proposer_task.worker_tag = self.WorkerTag.GENERATION
+
+        probe_task = GenerationTask()
+        probe_task.max_tokens = 20
+        probe_task.temperature = 0.6
+        probe_task.top_p = 0.95
+        probe_task.worker_tag = self.WorkerTag.GENERATION
+
+        probe_answers = []
+        probe_responses = []
+
+        initial_prompt_token_num = len(
+            self.tokenizer.encode(initial_prompt, add_special_tokens=False))
+        probe_suffix_token_num = len(
+            self.tokenizer.encode(self.probe_suffix, add_special_tokens=False))
+
+        current_prompt = initial_prompt
+```
+
+To prevent extra latency, the `proposer_task` should not be blocked by the `probe_task`. Scaffolding's task-level concurrency handles this perfectly. We can yield `proposer_task` and `probe_task` in a single list. Multiple tasks yielded together in the same list will be batched and executed in parallel.
+
+```python
+    yield[proposer_task, probe_task]
+```
+
+In the following `for` loop, each iteration performs these steps:
+
+1. **Submit** both a proposer task and a probe task by yielding them. We don't need to worry about execution details, as they are handled by `ScaffoldingLlm`, which binds the `Controller` and `Workers` together behind the scenes.
+
+2. **Evaluate** the probe response after the tasks return, checking for consistency over several rounds (using `certainty_threshold`).
+
+3. **Finalize** the answer and return if it is consistent. Otherwise, append the new tokens from the proposer task and proceed to the next iteration.
+
+```python
+        # Iterate over generation rounds until the maximum tokens limit is reached.
+        # Make sure length of prefilling is always smaller than the max_tokens in TRTLLMWorker.init_with_new_llm
```

Comment:
suggestion: "prefilling is always smaller than the max_tokens" -> "prefilling length is always less than max_tokens"